arangod/RestHandler/RestBatchHandler.cpp

```diff
@@ -351,14 +351,20 @@ bool RestBatchHandler::getBoundaryHeader(std::string* result) {
     return false;
   }
 
-  std::string boundary = "--" + parts[1].substr(boundaryLength);
+  std::string boundary = parts[1].substr(boundaryLength);
 
-  if (boundary.size() < 5) {
+  if ((boundary[0]  == '"') && (boundary[boundary.length() -1] == '"')) {
```

Comment:
this is potentially unsafe, e.g. when boundary is an empty string or a string of length 1.